server/src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -295,7 +296,8 @@ protected AsyncShardFetch.FetchResult<NodeStoreFilesMetadata> fetchData(ShardRou
                     logger,
                     "shard_store",
                     shard.shardId(),
-                    IndexMetadata.INDEX_DATA_PATH_SETTING.get(allocation.metadata().index(shard.index()).getSettings())
+                    IndexMetadata.INDEX_DATA_PATH_SETTING.get(allocation.metadata().index(shard.index()).getSettings()),
+                    allocation.nodes().getSize()
```

Comment:
Just the data nodes:

```suggestion
                    allocation.routingNodes().size()
```